app/src/main/java/io/apicurio/registry/auth/TokenRoleProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.auth;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.quarkus.security.identity.SecurityIdentity;
+
+/**
+ * @author eric.wittmann@gmail.com
+ */
+@ApplicationScoped
+public class TokenRoleProvider implements RoleProvider {
+
+    @Inject
+    AuthConfig authConfig;
+
+    @Inject
+    SecurityIdentity securityIdentity;
+
+    private boolean hasRole(String role) {
+        return securityIdentity.hasRole(role);
+    }
+
+    /**
+     * @see io.apicurio.registry.auth.RoleProvider#isAdmin()
+     */
+    @Override
+    public boolean isAdmin() {
```

Comment:
As a minor comment, I'm not sure if I like the idea of injecting the `SecurityIdentity` here. Wdyt about passing it as a param? Or possibly even better, pass a list of roles as a param so this is agnostic to the platform that is being used.